api/src/user/dto/query.dto.ts

```diff
@@ -0,0 +1,26 @@
+import { IsEmail, IsInt, IsOptional, IsString } from 'class-validator';
+import { Type } from 'class-transformer';
+
+export class QueryDto {
+  @IsString()
+  @IsOptional()
+  id?: string;
+
+  @IsString()
+  @IsOptional()
+  username?: string;
+
+  @IsEmail()
+  @IsOptional()
+  email?: string;
+
+  @IsInt()
+  @Type(() => Number)
+  @IsOptional()
+  page = 10;
```

Comment:
```suggestion
  page? = 10;
```

same for `page_number` below?

Reply:
I think it's fine. Giving it a default value is enough to make it optional, we don't need the `?`.